packages/block-editor/src/components/inner-blocks/README.md

```diff
@@ -90,6 +90,69 @@ const TEMPLATE = [ [ 'core/columns', {}, [
 
 The previous example creates an InnerBlocks area containing two columns one with an image and the other with a paragraph.
 
+### `templateOptions`
```

Comment:
Honestly, I'm still uncertain about this prop. It seems like that something that could have easily been built in the columns block itself (not render `InnerBlocks` until a template is chosen).

Is the idea here to "generalize" this behavior? Is this part of the goal of the original issue? 

I can see value just trying to see if we really want to introduce it now, or maybe try to see if we need it in at least two blocks to introduce something like that.

Reply:
> Honestly, I'm still uncertain about this prop. It seems like that something that could have easily been built in the columns block itself (not render `InnerBlocks` until a template is chosen).
> 
> Is the idea here to "generalize" this behavior? Is this part of the goal of the original issue?

From the original comment:

>I'd appreciate feedback about whether this is a reasonable enhancement for InnerBlocks, or if we should consider shifting this responsibility back to the block component to handle (optionally enhancing Placeholder instead, or creating new components to render a set of Button choices).

From the related issue (and as evidenced by prior art in a number of plugins), there's a desire for a common behavior here in template selection. I worry about inconsistency if we start proposing that it be implemented ad hoc by each block.

As in the text above, I think there might be some way we can still put it on the block to implement, with sufficient component coverage (`Placeholder`) to _at least strongly encourage_ conforming to a consistent UI.

Reply:
I think I worry more about the growing number of props the `InnerBlocks` could and will have in the future. Maybe a first step here could be to make the `TemplatePicker` component, the sharable API, the same way we have `MediaPlaceholder`...

Reply:
Sorry, I arrived late to the party. I like the idea of providing a consistent UI for this scenario that we predict may become very common. I share the concern of increasing the number of props/responsibilities of InnerBlocks with @youknowriad.
What if we only add a new prop called placeholder to InnerBlocks (it just renders a placeholder when the template is null and no child blocks exist). Then we can expose a TemplatePlaceholder component that handles all the UI? (I think this version was exactly what @youknowriad suggested). It seems InnerBlocks should not receive a prop that is called when a template is selected and it may be good to extract that behavior. Even having a placeholder prop that receives an element rendered when the placeholder should be shown is an unnecessary prop but it is useful otherwise blocks would need to check the number of child blocks they have and conditionally render the placeholder.